src/client/executor.rs

```diff
@@ -108,7 +108,7 @@ impl Client {
 
                 // Retryable writes are only supported by storage engines with document-level
                 // locking, so users need to disable retryable writes if using mmapv1.
-                if let ErrorKind::CommandError(ref err) = err.kind.as_ref() {
+                if let ErrorKind::CommandError(ref err) = err.kind {
```

Comment:
is there any reason we would want `kind` to be a method instead of accessing the field directly? This is what [`std::io::Error` does](https://doc.rust-lang.org/std/io/struct.Error.html#method.kind), but I don't really have any further justification than that.

Reply:
The main benefit would be that we would be able to change the way that we store the field in the struct without breaking API. As it is currently, removing or changing the type of the field would break user code that accesses the field directly.

Reply:
I don't really have any strong feelings on this either way, if you think that protection is worthwhile then I'd say go ahead and make the change and if not, seems fine to me to leave it as is.